### PR TITLE
Create quick_sort.rb

### PR DESCRIPTION
**Pull request by cy-by-side**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [x] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [x] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [x] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

File: codes/ruby/chapter_sorting/quick_sort.rb [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      if pivot - left < right - pivot
+        quick_sort(nums, left, pivot - 1)
+        left = pivot + 1 # 剩余未排序区间为 [pivot + 1, right]
+      else
+        quick_sort(nums, pivot + 1, right)
+        right = pivot - 1 # 剩余未排序区间为 [left, pivot - 1]
+      end
+    end
+  end
+  end
```
Comment (krahets) [inline codes/ruby/chapter_sorting/quick_sort.rb:124]: Incorrect indentation

**Comment by krahets** [comment]
Thanks for the PR @cy-by-side! The Ruby code in this project follows the [ruby style guide](https://github.com/JuanitoFatas/ruby-style-guide/blob/master/README.md), so please check your code against it.

**Comment by krahets** [comment]
@khoaxuantu Could you help review #1223 #1222 #1221 to preserve the consistency of Ruby code? Thx!

File: codes/ruby/chapter_sorting/quick_sort.rb (hunk 1 above)
Comment (krahets) [reply codes/ruby/chapter_sorting/quick_sort.rb:124]: Recommend formatting the code before submitting the PR.

File: codes/ruby/chapter_sorting/quick_sort.rb [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+### Driver Code ###
+
+# 快速排序
+nums = [2, 4, 1, 0, 3, 5]
+QuickSort.quick_sort(nums, 0, nums.length - 1)
+puts "快速排序完成后 nums = #{nums}"
+
+nums1 = [2, 4, 1, 0, 3, 5]
```
Comment (khoaxuantu) [inline codes/ruby/chapter_sorting/quick_sort.rb:133]: Please add this comment to above `nums1`

```rb
# 快速排序（中位基准数优化）
```

File: codes/ruby/chapter_sorting/quick_sort.rb [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+QuickSortMedian.quick_sort(nums1, 0, nums1.length - 1)
+puts "快速排序（中位基准数优化）完成后 nums1 = #{nums1}"
+
+nums2 = [2, 4, 1, 0, 3, 5]
```
Comment (khoaxuantu) [inline codes/ruby/chapter_sorting/quick_sort.rb:137]: Please add this comment to above `nums2`

```rb
# 快速排序（尾递归优化）
```

**Review by khoaxuantu — COMMENTED** [review]
Hi @cy-by-side, I have some more comments

File: codes/ruby/chapter_sorting/quick_sort.rb [hunk 4]
```diff
@@ -0,0 +1,153 @@
+=begin
+File: quick_sort.rb
+Created Time: 2024-04-01
+Author: Cy (3739004@gmail.com) Xuan Khoa Tu Nguyen (ngxktuzkai2000@gmail.com)
```
Comment (krahets) [inline codes/ruby/chapter_sorting/quick_sort.rb:4]: Add a comma between the two ids.

**Comment by cy-by-side** [comment]
does it be able to merge?

File: codes/ruby/chapter_sorting/quick_sort.rb [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+# 快速排序（中位基准数优化）
+nums1 = [2, 4, 1, 0, 3, 5]
+QuickSortMedian.quick_sort(nums1, 0, nums1.length - 1)
+puts "快速排序（中位基准数优化）完成后 nums1 = #{nums1}"
+
+# 快速排序（尾递归优化）
+nums2 = [2, 4, 1, 0, 3, 5]
+QuickSortTailCall.quick_sort(nums2, 0, nums2.length - 1)
+puts "快速排序（尾递归优化）完成后 nums2 = #{nums2}"
```
Comment (khoaxuantu) [inline codes/ruby/chapter_sorting/quick_sort.rb:153]: I think it is almost done now, just one more thing. Can you add the driver code into this `if __FILE__ == $0` block.

```suggestion
if __FILE__ == $0
  # 快速排序
  nums = [2, 4, 1, 0, 3, 5]
  QuickSort.quick_sort(nums, 0, nums.length - 1)
  puts "快速排序完成后 nums = #{nums}"

  # 快速排序（中位基准数优化）
  nums1 = [2, 4, 1, 0, 3, 5]
  QuickSortMedian.quick_sort(nums1, 0, nums1.length - 1)
  puts "快速排序（中位基准数优化）完成后 nums1 = #{nums1}"

  # 快速排序（尾递归优化）
  nums2 = [2, 4, 1, 0, 3, 5]
  QuickSortTailCall.quick_sort(nums2, 0, nums2.length - 1)
  puts "快速排序（尾递归优化）完成后 nums2 = #{nums2}"
end
```

**Review by khoaxuantu — APPROVED** [review]

**Review by krahets — APPROVED** [review]
Thanks for the PR!

**Pull request merged**